Library/Client.ts

```diff
@@ -145,14 +145,44 @@ class Client {
         this.track(data, tagOverrides, contextObjects);
     }
 
+    /**
+     * Log an incoming request. This method will synchronously record the event, rather than waiting for response send.
+     * Use this when you need to perform custom logic to record the duration of the request.
+     * 
+     * This call will also add outgoing headers to the supplied response object for correlating telemetry across different services.
+     * @param request              http.ServerRequest - the request object to monitor
+     * @param response             http.ServerResponse - the response object to monitor
+     * @param elapsedMilliseconds  number - the elapsed time taken to handle this request in milliseconds
+     * @param properties           map[string, string] - additional data used for filtering in the portal. Defaults to empty.
+     * @param error                any - an object indicating the request was unsuccessful. This object will be recorded with the request telemetry.
+     */
     public trackRequestSync(request: http.ServerRequest, response: http.ServerResponse, ellapsedMilliseconds?: number, properties?: { [key: string]: string; }, error?: any) {
```

Comment:
Should we have `startRequest` `stopRequest` instead? `Sync` is really misleading in this context.